kubernetes/cache/kube_cache.go

```diff
@@ -1467,12 +1467,27 @@ func (c *kubeCache) GetTelemetries(namespace, labelSelector string) ([]*v1alpha1
 }
 
 func (c *kubeCache) isK8sGatewayListerInit(namespace string) bool {
-	// K8s GW has several cache listers, those can be namespace or cluster based
-	// if one of them is not initialized, then all others are not init as well
-	// this can happen when CRDs are created after Kiali start
-	if c.getCacheLister(namespace).k8sgatewayLister == nil {
```

Comment:
Looks like the issue is that `c.getCacheLister(namespace)` can return `nil`? Would probably be simpler to just add a conditional here then that checks for that before accessing `k8sgatewayLister`. Something like:
```
if cacheLister := c.getCacheLister(namespace); cacheLister == nil || cacheLister.k8sgatewayLister == nil {
  return false
}
```

Reply:
then it will always return false on OCP cluster and the Gateway functionality could be missed

Reply:
yeah agree, I had that idea on my previous PR, but avoided adding a new variable, now it sound better idea than many checks